genesiscommunitysuccess/custom-elements-lsp
# feat: refactor into packages and publish skeleton analyzer plugin FUI-1512

📓  &nbsp; **Related Issue**
If this PR is related to an issue then link it here, or delete this section.

🤔  &nbsp; **What does this PR do?**

Restructure the project as a monorepo and use lerna to publish multiple packages
- Restructure the project using `npm workspaces` 
- Enable lerna for publishing

📑  &nbsp; **How should this be tested?**

Update test steps to match your PR:

```
1. Checkout branch
2. `npm run bootstrap`
3. `npm run test:unit`

To test the LSP is working correctly in vscode
1. `cd packages/showcase/example`
2. `code .` 
3. Open a typescript file such as `root.ts`
4. In the Command Palette choose `TypeScript: select typescript version...` and then choose the workspace version
5. The LSP should be enabled. You will need to make a code change after the LSP is enabled before you see any Intellisense.
```

> These testing instructions assume that you've already setup the LSP in your IDE with the `example` app. If you haven't then follow the instructions in the `README.md`.

✅  &nbsp; **Checklist**
